src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -387,65 +379,87 @@ public OperationQueue<TOperation> Stop()
                 State = QueueState.Stopped;
                 return result;
             }
-        }
 
-        private static SocketAsyncContext s_closedAsyncContext;
-        public static SocketAsyncContext ClosedAsyncContext
-        {
-            get
+            public void Complete(SocketAsyncContext context)
             {
-                if (Volatile.Read(ref s_closedAsyncContext) == null)
+                if (IsStopped)
+                    return;
+
+                State = QueueState.Set;
+
+                TOperation op;
+                while (TryDequeue(out op))
                 {
-                    var ctx = new SocketAsyncContext(-1, null);
-                    ctx.Close();
+                    if (!op.TryCompleteAsync(context))
+                    {
+                        Requeue(op);
```

Comment:
Is there any risk that this effectively ends up being an infinite loop? i.e. call Complete, it dequeues an operation, TryCompleteAsync fails, we requeue and return, ... repeat?  I haven't looked at how this is consumed, so just asking; it's probably fine.